### DISCUSSION TITLE
Negative "Percentage Surcharge" is considered as a positive one

### DISCUSSION THREAD

**Discussion by langerhorst** [mampfes/ha_epex_spot#66] [Q&A]
THANK you for that wonderful integration at first. 
I tried to fill in the numbers in "Configure surcharges and tax to calculate net price per kWh."
We do have a negative "percentage surcharge". As I filled in -4% the algorithm behind interpreted this as a positive surcharge. But I need a negative one. Is it possible to fix that? This way, I would be able to calculate the whole price with nearly everything in it. 
Our electricity supplier has an absolute surcharge of 3.8 Cent but in addition gives us back 4%. So we would need this negative surcharge to have the right price.

**Comment by mampfes** [comment]
Hi,
if I got you right, then the following change should do the trick:
https://github.com/mampfes/ha_epex_spot/blob/main/custom_components/epex_spot/SourceShell.py#L135
```
< net_p = net_p + abs(net_p * surcharge_pct / 100)
> net_p = net_p + abs(net_p) * surcharge_pct / 100
```
Could you please test it?

**Comment by mampfes** [comment]
Fixed in release 2.2.0

**Comment by langerhorst** [comment]
I updated the EPEX plugin and found out, that there are still far to high prices. 
See the attached Screenshots.
The prices should be divided by 100 - then the seem to be right. 
I use aWATTar as source for the prices.

![Bildschirmfoto vom 2023-11-28 19-18-03](https://github.com/mampfes/ha_epex_spot/assets/70429589/4baa2b42-d970-42df-94cc-2d92740142c3)
![Bildschirmfoto vom 2023-11-28 19-15-31](https://github.com/mampfes/ha_epex_spot/assets/70429589/3994fb12-2a00-4e71-b8d0-45039ee8daad)
![Bildschirmfoto vom 2023-11-28 19-14-47](https://github.com/mampfes/ha_epex_spot/assets/70429589/63b6b4f3-e8be-45f5-892b-739bf3897c4d)

Thanks for the quick fix, maybe I configured something wrong?
"EPEX Spot Data Net Price" VS "EPEX Spot Data Price" --> I took the first one...

**Reply by mr-p666** [reply]
EPEX Spot Data Net Price is in Cent/kWh but you need EUR/kWh (as u can see in your screenshot).
I created a helper for that.

**Comment by langerhorst** [comment]
Negative Surcharge works perfect. Now I added a helper, that the price from aWATTar API converts from Cent/kWh to EUR/kWh. Tomorrow I'll see if that works. Thanks for this advice :)
